### PR TITLE
Integrate typed nats-protoo

### PR DESCRIPTION
**Pull request by jbrady42**
Add types to internal messaging.

Also add type aliases for UID MID RID for extra safety in case these change in the future.

Still some work to be done on typing all messages. Parsing to `map[string]interface{}` is still supported and used to bridge the gap for the meantime.

### CODE REVIEW

File: pkg/node/islb/internal.go [hunk 1]
```diff
@@ -263,67 +242,69 @@ func getPubs(data map[string]interface{}) (map[string]interface{}, *nprotoo.Erro
...
 			}
 		}
-		pub := util.Map("rid", rid, "uid", info.UID, "mid", info.MID, "info", fields["info"], "tracks", tracks)
+
+		var extraInfo proto.ClientUserInfo
+		if err := json.Unmarshal([]byte(fields["info"]), &extraInfo); err != nil {
+			log.Errorf("Unmarshal pub extra info %v", err)
+			extraInfo = proto.ClientUserInfo{}
+		}
+		// pub := util.Map("rid", rid, "uid", info.UID, "mid", info.MID, "info", fields["info"], "tracks", tracks)
```
Comment (tarrencev) [inline pkg/node/islb/internal.go]: cleanup

File: pkg/node/islb/internal.go [hunk 2]
```diff
@@ -263,67 +242,69 @@ func getPubs(data map[string]interface{}) (map[string]interface{}, *nprotoo.Erro
...
 		pubs = append(pubs, pub)
 	}
 
-	resp := util.Map("rid", rid, "uid", uid)
-	resp["pubs"] = pubs
+	resp := proto.GetPubResp{
+		RoomInfo: data,
+		Pubs:     pubs,
+	}
+	// util.Map("rid", rid, "uid", uid)
```
Comment (tarrencev) [inline pkg/node/islb/internal.go]: cleanup

File: pkg/node/islb/internal.go [hunk 3]
```diff
@@ -383,49 +364,70 @@ func unRelay(data map[string]interface{}) (map[string]interface{}, *nprotoo.Erro
...
-	uid := util.Val(data, "uid")
-	info := util.Val(data, "info")
-	msg := util.Map("rid", rid, "uid", uid, "info", info)
-	log.Infof("broadcaster.Say msg=%v", msg)
-	broadcaster.Say(proto.IslbOnBroadcast, msg)
-	return util.Map(), nil
+func broadcast(data proto.BroadcastMsg) (interface{}, *nprotoo.Error) {
+	log.Infof("broadcaster.Say msg=%v", data)
+	broadcaster.Say(proto.IslbOnBroadcast, data)
+	return struct{}{}, nil
```
Comment (tarrencev) [inline pkg/node/islb/internal.go:373]: remove return types?

File: pkg/node/islb/internal.go [hunk 4]
```diff
@@ -362,10 +343,10 @@ func relay(data map[string]interface{}) (map[string]interface{}, *nprotoo.Error)
 		log.Infof("amqp.RpcCall ip=%s, method=%v", ip, method)
 		//amqp.RpcCall(ip, method, "")
 	}
-	return util.Map(), nil
+	return struct{}{}, nil
```
Comment (tarrencev) [inline pkg/node/islb/internal.go:349]: remove return types?

File: pkg/node/islb/internal.go [hunk 5]
```diff
@@ -263,67 +242,69 @@ func getPubs(data map[string]interface{}) (map[string]interface{}, *nprotoo.Erro
...
 	}
-	msg := util.Map("rid", rid, "uid", uid)
-	log.Infof("Broadcast peer-leave = %v", msg)
+	log.Infof("Broadcast peer-leave = %v", data)
 	//make broadcast leave msg after remove stream msg, for ion block bug
 	time.Sleep(500 * time.Millisecond)
-	broadcaster.Say(proto.IslbClientOnLeave, msg)
-	return util.Map(), nil
+	broadcaster.Say(proto.IslbClientOnLeave, data)
+	return struct{}{}, nil
```
Comment (tarrencev) [inline pkg/node/islb/internal.go:303]: Seems a bunch of these methods don't return anything useful. Should we just remove the return types?

File: pkg/node/sfu/internal.go [hunk 6]
```diff
@@ -126,17 +127,22 @@ func publish(msg map[string]interface{}) (map[string]interface{}, *nprotoo.Error
 		return nil, util.NewNpError(415, "publish: pub.Answer failed.")
 	}
 
-	router.AddPub(uid, pub)
+	router.AddPub(string(uid), pub)
```
Comment (tarrencev) [inline pkg/node/sfu/internal.go]: Should we change these methods to accept `UID`? If so, maybe add a `TODO` here if we want to do that separately.

**Review by tarrencev — APPROVED** [review]
Awesome stuff. Much cleaner!

File: pkg/node/biz/internal.go [hunk 7]
```diff
@@ -17,26 +20,30 @@ func strToMap(msg map[string]interface{}, key string) {
...
+	var isblSignalTransformMap = map[string]string{
+		proto.IslbOnStreamAdd:    proto.ClientOnStreamAdd,
+		proto.IslbOnStreamRemove: proto.ClientOnStreamRemove,
+		proto.IslbClientOnJoin:   proto.ClientOnJoin,
+		proto.IslbClientOnLeave:  proto.ClientOnLeave,
+		proto.IslbOnBroadcast:    proto.ClientBroadcast,
+	}
+	go func(msg nprotoo.Notification) {
+		var data proto.BroadcastMsg
+		if err := json.Unmarshal(msg.Data, &data); err != nil {
```
Comment (jbrady42) [inline pkg/node/biz/internal.go]: Use `msg.Data.Unmarshal`

File: pkg/node/islb/internal.go (hunk 5 above)
Comment (jbrady42) [reply pkg/node/islb/internal.go:303]: The empty return is to meet the Request interface of handling a message and returning a response. My hope is to eventually collapse the request handling switch, in which case the return type is still required.

**Pull request merged**